From a75a27dfde2d4668c37ed3bb226c66fd55b2f003 Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Tue, 20 Jun 2023 07:04:03 +0000
Subject: [PATCH] [wasm][cmake] HACK: link wasi-emulated-mman to host
 executable

The reason why this change is needed is unclear to me...
---
 cmake/modules/AddSwift.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/modules/AddSwift.cmake b/cmake/modules/AddSwift.cmake
index e0fb5de09f2..834138ba8f6 100644
--- a/cmake/modules/AddSwift.cmake
+++ b/cmake/modules/AddSwift.cmake
@@ -397,6 +397,8 @@ function(_add_host_variant_link_flags target)
       cxx_link_libraries)
     target_link_libraries(${target} PRIVATE
       ${cxx_link_libraries})
+  elseif("${LFLAGS_SDK}" STREQUAL "WASI")
+    list(APPEND result "-Wl,wasi-emulated-mman")
   else()
     # If lto is enabled, we need to add the object path flag so that the LTO code
     # generator leaves the intermediate object file in a place where it will not
-- 
2.40.0

